src/Commands/Setup.php

```diff
@@ -0,0 +1,236 @@
+     *
+     * @param string $file Relative file path like 'Config/Auth.php'.
+     */
+    protected function writeFile(string $file, string $content)
+    {
+        $config  = new Autoload();
+        $appPath = $config->psr4[APP_NAMESPACE];
+
+        $path      = $appPath . $file;
+        $cleanPath = clean_path($path);
+
+        $directory = dirname($path);
+
+        if (! is_dir($directory)) {
+            mkdir($directory, 0777, true);
+        }
+
+        if (file_exists($path)) {
+            $overwrite = (bool) CLI::getOption('f');
+
+            if (
+                ! $overwrite
+                && CLI::prompt("  File '{$cleanPath}' already exists in destination. Overwrite?", ['n', 'y']) === 'n'
+            ) {
+                CLI::error("  Skipped {$cleanPath}. If you wish to overwrite, please use the '-f' option or reply 'y' to the prompt.");
+
+                return;
+            }
+        }
+
+        if (write_file($path, $content)) {
+            CLI::write(CLI::color('  Created: ', 'green') . $cleanPath);
+        } else {
+            CLI::error("  Error creating {$cleanPath}.");
+        }
+    }
+
+    /**
+     * @param string $check Code to add.
+     * @param string $file  Relative file path like 'Controllers/BaseController.php'.
+     */
+    protected function replace(string $file, string $check, string $pattern, string $replace)
+    {
+        $config    = new Autoload();
+        $appPath   = $config->psr4[APP_NAMESPACE];
+        $path      = $appPath . $file;
+        $cleanPath = clean_path($path);
+
+        $content = file_get_contents($path);
+
+        $return = preg_match('/' . preg_quote($check, '/') . '/u', $content);
+        if ($return === 1) {
+            CLI::error("  Skipped {$cleanPath}. It has already been updated.");
+
+            return;
+        }
+        if ($return === false) {
+            CLI::error("  Error checking {$cleanPath}.");
+
+            return;
+        }
+
+        $content = preg_replace($pattern, $replace, $content);
+
+        if (write_file($path, $content)) {
+            CLI::write(CLI::color('  Updated: ', 'green') . $cleanPath);
+        } else {
+            CLI::error("  Error updating {$cleanPath}.");
+        }
+    }
+
+    private function setupHelper()
+    {
+        $file = 'Controllers/BaseController.php';
+
+        $check   = '$this->helpers = array_merge($this->helpers, [\'auth\', \'setting\']);';
+        $pattern = '/(' . preg_quote('// Do Not Edit This Line', '/') . ')/u';
+        $replace = $check . "\n\n        " . '$1';
+
+        $this->replace($file, $check, $pattern, $replace);
+    }
+
+    private function setupRoutes()
+    {
+        $file = 'Config/Routes.php';
+
+        $check   = 'service(\'auth\')->routes($routes);';
+        $pattern = '/(.*)(\n' . preg_quote('$routes->', '/') . '[^\n]+?\n)/su';
+        $replace = '$1$2' . "\n" . $check . "\n";
+
+        $this->replace($file, $check, $pattern, $replace);
+    }
```

Comment:
private function setupMigrates(): void
    {
        // Call those commands!
        command('migrate -n CodeIgniter\\\\Settings\\\\');
        command('migrate -n CodeIgniter\\\\Shield\\\\');
    }

Reply:
I like adding migrations to this but probably would make more sense to use the class directly instead of using `command()`.

Reply:
I agree, it was logical.

Reply:
Done.